src/signTxHash.c

```diff
@@ -197,21 +197,24 @@ void handleSignTxHash(uint8_t p1, uint8_t *dataBuffer, uint16_t dataLength, vola
     }
 
     // sign the hash
-    if (!sign_tx_hash(dataBuffer)) {
+    if (!sign_tx_hash(data_buffer)) {
         init_tx_context();
         THROW(ERR_SIGNATURE_FAILED);
     }
 
     bool should_display_esdt_flow = false;
-    if ((esdt_info.identifier_len > 0) &&
-        (strncmp(tx_context.data + DATA_SIZE_LEN - 1, ESDT_TRANSFER_PREFIX, strlen(ESDT_TRANSFER_PREFIX)) == 0) &&
-        (strncmp(tx_context.data + DATA_SIZE_LEN - 1 + strlen(ESDT_TRANSFER_PREFIX), esdt_info.identifier, esdt_info.identifier_len) == 0) &&
-        (strncmp(tx_context.chain_id, esdt_info.chain_id, MAX_CHAINID_LEN) == 0)) {
+    bool identifier_len_valid = esdt_info.identifier_len > 0;
+    bool has_data = strlen(tx_context.data) > 0;
+    bool has_esdt_transfer_prefix = strncmp(tx_context.data + DATA_SIZE_LEN - 1, ESDT_TRANSFER_PREFIX, strlen(ESDT_TRANSFER_PREFIX)) == 0;
+    bool has_registered_esdt_identifier = strncmp(tx_context.data + DATA_SIZE_LEN - 1 + strlen(ESDT_TRANSFER_PREFIX), esdt_info.identifier, esdt_info.identifier_len) == 0;
+    bool same_chainid = strncmp(tx_context.chain_id, esdt_info.chain_id, MAX_CHAINID_LEN) == 0;
+
+    if (has_data && identifier_len_valid && has_esdt_transfer_prefix && has_registered_esdt_identifier && same_chainid) {
```

Comment:
Maybe change the code here to avoid some possible undefined behavior.
`esdt_info.identifier_len` and `tx_context.data` are used without being checked if they are valid.  Check `has_data` and `identifier_len_valid` before using structure fields 

```suggestion
    bool identifier_len_valid = esdt_info.identifier_len > 0;
    bool has_data = strlen(tx_context.data) > 0;
    
    if (has_data && identifier_len_valid)
    {
        bool has_esdt_transfer_prefix = strncmp(tx_context.data + DATA_SIZE_LEN - 1, ESDT_TRANSFER_PREFIX, strlen(ESDT_TRANSFER_PREFIX)) == 0;
         bool has_registered_esdt_identifier = strncmp(tx_context.data + DATA_SIZE_LEN - 1 + strlen(ESDT_TRANSFER_PREFIX), esdt_info.identifier, esdt_info.identifier_len) == 0;
         bool same_chainid = strncmp(tx_context.chain_id, esdt_info.chain_id, MAX_CHAINID_LEN) == 0;
    
         if (has_esdt_transfer_prefix && has_registered_esdt_identifier && same_chainid) {
    }
    
```